src/forms/Form/validators.ts

```diff
@@ -13,14 +13,16 @@ export function validateEmail(str: string): string | undefined {
 	if (!str) {
 		return;
 	}
-
-	const isValidEmail = str
-		.toLowerCase()
-		.match(
-			/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
-		);
-
-	if (isValidEmail) return;
+	
+	if (typeof str === "string") {
+		const isValidEmail = str
+			.toLowerCase()
+			.match(
+				/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+			);
+	
+		if (isValidEmail) return;
+	}
 
 	return "The value is not a valid e-mail";

```

Comment:
The error should only appear if the value is a string. The idea of this ticket is to only show error messages on the fields they belong, so for example, this checkbox component shouldn't show the error message
![image](https://user-images.githubusercontent.com/23409284/166839454-97152c61-145b-4dd8-8ee6-342d2a32e9c5.png)